File: src/main/java/com/esprit/stage/Service/StatistiqueService.java
package com.esprit.stage.Service;

import com.esprit.stage.Repository.EvaluationRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StatistiqueService {

    @Autowired
    private EvaluationRepository evaluationRepository;

    // Get the average global rating
    public Double getMoyenneNoteGlobale() {
        return evaluationRepository.getMoyenneNoteGlobale();
    }

    // Get the average quality of supervision
    public Double getMoyenneQualiteEncadrement() {
        return evaluationRepository.getMoyenneQualiteEncadrement();
    }

    // Get the average work atmosphere
    public Double getMoyenneAmbianceTravail() {
        return evaluationRepository.getMoyenneAmbianceTravail();
    }

    // Get the total number of complaints
    public Long getTotalComplaints() {
        return evaluationRepository.count();
    }

    // Get the workload distribution
    public List<Object[]> getRepartitionChargeTravail() {
        return evaluationRepository.getRepartitionChargeTravail();
    }

    // Get the complaints distribution
    public List<Object[]> getRepartitionPlaintes() {
        return evaluationRepository.getRepartitionPlaintes();
    }

    // Get comment distribution


    // Get details distribution
    public List<Object[]> getRepartitionDetails() {
        return evaluationRepository.getRepartitionDetails();
    }
}

File: src/app/services/statistique.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class StatistiqueService {

  private baseUrl: string = 'http://localhost:8888/api/statistics';

  constructor(private http: HttpClient) {}

  // Fetch average global ratings
  getMoyenneNoteGlobale(): Observable<number> {
    return this.http.get<number>(`${this.baseUrl}/moyenne-note`);
  }

  // Fetch average supervision quality
  getMoyenneQualiteEncadrement(): Observable<number> {
    return this.http.get<number>(`${this.baseUrl}/moyenne-qualite`);
  }

  // Fetch average work atmosphere
  getMoyenneAmbianceTravail(): Observable<number> {
    return this.http.get<number>(`${this.baseUrl}/moyenne-ambiance`);
  }

  // Fetch total number of complaints
  getTotalComplaints(): Observable<number> {
    return this.http.get<number>(`${this.baseUrl}/total-complaints`);
  }

  // Fetch workload distribution
  getRepartitionChargeTravail(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/charge-travail`);
  }

  // Fetch complaints distribution
  getRepartitionPlaintes(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/repartition-plaintes`);
  }

  // Fetch comment distribution
  getRepartitionCommentaires(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/repartition-commentaires`);
  }

  // Fetch detail distribution
  getRepartitionDetails(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/repartition-details`);
  }
}
